### ISSUE TITLE
Validate openapi specs

### ISSUE THREAD

**Issue by Leptopoda** [nextcloud/neon#1131]
We should validate our specs with an openapi validator.
This is especially useful for hand written specs in the dynamite_e2e tests and could avoid things like #1098
It could also make us more confident when applying patches to the generated specs.

**Comment by provokateurin** [comment]
We could also add a script in bin/ which developers can use to validate their specs (without generating any code).

**Comment by Leptopoda** [comment]
Shure.
What tool do you think fits best? I only know the `openapi-builder` that can also validate but I have little trust in that software.

**Comment by provokateurin** [comment]
No I meant that we can build our own based on our spec parsing :)

**Comment by Leptopoda** [comment]
We already partially do so when generating.
But a bug in the parsing for dynamite would not help us validate the e2e specs.
If they are wrong we are testing dynamite with garbage data.

**Comment by provokateurin** [comment]
Of course, I mean it might just be nice to expose this separately from generating code.
